pkg/controller/common/helpers.go

```diff
@@ -765,6 +770,30 @@ func CalculateConfigFileDiffs(oldIgnConfig, newIgnConfig *ign3types.Config) []st
 	return diffFileSet
 }
 
+// NewIgnFile returns a simple ignition3 file from just path and file contents.
+// It also ensures the compression field is set to the empty string, which is
+// currently required for handling config merging correctly.
+//
+// Note the default Ignition file mode is 0644, owned by root/root.
+func NewIgnFile(path, contents string) ign3types.File {
+	return NewIgnFileBytes(path, []byte(contents))
+}
+
+// NewIgnFileBytes is like NewIgnFile, but accepts binary data
+func NewIgnFileBytes(path string, contents []byte) ign3types.File {
+	return ign3types.File{
+		Node: ign3types.Node{
+			Path: path,
+		},
+		FileEmbedded1: ign3types.FileEmbedded1{
+			Contents: ign3types.Resource{
+				Source:      StrToPtr(dataurl.EncodeBytes(contents)),
```

Comment:
See below

Reply:
Well, hmm...actually it does mean that these files will start appearing as base64 inside MachineConfig, which will mean people need to base64 decode them manually if relevant.

It will also be a "spurious" change on upgrades, e.g. the machineconfig hash will be different, but the physical files will not.

I am not sure if either of these really matter though.